src/psyclone/tests/psyir/transformations/hoist_trans_test.py

```diff
@@ -393,7 +392,91 @@ def test_validate_dependencies_if_statement(fortran_reader):
             "the loop, outside the supplied statement." in str(err.value))
 
 
-# str
+def test_validate_checks_for_side_effects(fortran_reader):
+    ''' Test that the validate refuses cases that may contain side effects
+    such as CodeBlocks or Call not guaranteed to be pure.
+    '''
+    psyir = fortran_reader.psyir_from_source(
+        '''
+        subroutine test()
+          use other
+          do i=1,10
+              write(zchar1, '(I2.2)') i
```

Comment:
No need for this line?

Reply:
Technically no, because now it will not hoist the CodeBlock, but I'd like to keep it because it shows a case where hoisting it would change the meaning. Otherwise we could thing that hoisting the second line is alright.